sys/arch/magic/Makefile

```diff
@@ -47,8 +47,8 @@ $(TARGET): $(OBJS) $(LIBKERNSTMP)
 ifeq ($(CROSS),yes)
 
 # for cross-compiling use this: (assumes you have perl and cnm)
-magic.i: $(TARGET)
-	perl genmagic.pl > $@
+magic.i: genmagic.o
+	perl genmagic.pl > $@.$$$$ && mv $@.$$$$ $@ || { $(RM) $@.$$$$; false; }
 
```

Comment:
That wizardry writes to a tmp file and moves it only when it finished successfully. Otherwise when using parallel makes, the redirection wil create it in one make, and the 2nd make will think it is up to date and continue with an empty output file because the first command isn't finished yet.